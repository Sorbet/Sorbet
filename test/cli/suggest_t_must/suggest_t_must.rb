# typed: true

foo = T.let(nil, T.nilable(String))
foo[0]

"hi" + foo

T::Array[T.nilable(Integer)].new.map(&:even?)
