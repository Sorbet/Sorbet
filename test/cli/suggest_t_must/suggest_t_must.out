test/cli/suggest_t_must/suggest_t_must.rb:4: Method `[]` does not exist on `NilClass` component of `T.nilable(String)` https://srb.help/7003
     4 |foo[0]
        ^^^^^^
  Got T.nilable(String) originating from:
    test/cli/suggest_t_must/suggest_t_must.rb:3:
     3 |foo = T.let(nil, T.nilable(String))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    test/cli/suggest_t_must/suggest_t_must.rb:4: Replace with `T.must(foo)`
     4 |foo[0]
        ^^^

test/cli/suggest_t_must/suggest_t_must.rb:6: Expected `String` but found `T.nilable(String)` for argument `arg0` https://srb.help/7002
     6 |"hi" + foo
        ^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/master/rbi/core/string.rbi#L58: Method `String#+` has specified `arg0` as `String`
    58 |        arg0: String,
                ^^^^
  Got T.nilable(String) originating from:
    test/cli/suggest_t_must/suggest_t_must.rb:3:
     3 |foo = T.let(nil, T.nilable(String))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  Autocorrect: Use `-a` to autocorrect
    test/cli/suggest_t_must/suggest_t_must.rb:6: Replace with `T.must(foo)`
     6 |"hi" + foo
               ^^^
Errors: 2
